spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/FunctionConfiguration.java

```diff
@@ -30,32 +32,32 @@
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.lang.Nullable;
 
 /**
- *
  * @author Oleg Zhurakousky
- *
+ * @author David Turanski
  * @since 2.1
  */
 @Configuration
 @ConditionalOnProperty("spring.cloud.stream.function.definition")
 public class FunctionConfiguration {
 
-	@Autowired(required=false)
+	@Autowired(required = false)
 	private Source source;
 
-	@Autowired(required=false)
+	@Autowired(required = false)
 	private Processor processor;
 
-	@Autowired(required=false)
+	@Autowired(required = false)
 	private Sink sink;
 
 	@Bean
 	public IntegrationFlowFunctionSupport functionSupport(FunctionCatalogWrapper functionCatalog,
```

Comment:
Can't we make this `static` as well and then use method argument injection in that new `integrationFlowFunctionSupportAwareBeanPostProcesser()` which should be `static` as well?

Reply:
That breaks some other things:  `Caused by: java.lang.IllegalArgumentException: 'functionCatalog' must not be null`  (Made that bean method static as well)